packages/babel-types/src/validators/react/isCompatTag.js

```diff
@@ -1,4 +1,5 @@
 // @flow
 export default function isCompatTag(tagName?: string): boolean {
-  return !!tagName && /^[a-z]|-/.test(tagName);
+  // Must start with 'a'-'z', can have a dash '-', but must finish with 'a'-'z'
+  return !!tagName && /^[a-z]+(-?[a-z]+)?$/.test(tagName);
```

Comment:
This pattern would reject custom element names with more than one hyphen (eg `my-custom-tag`).

I think this would work though?
`/^[^-]([\-a-z]*[^-])*$/`

Reply:
By the way, glancing at [the spec](https://html.spec.whatwg.org/multipage/custom-elements.html#valid-custom-element-name) it seems like a much broader range of characters than just a-z are allowed for custom elements (eg `<emotion-😍>`) 😆 Not sure if we want to try to support them all. Probably not?

Reply:
Yep the [custom elements spec](https://www.w3.org/TR/custom-elements/#valid-custom-element-name) says:

```
PotentialCustomElementName ::=
[a-z] (PCENChar)* '-' (PCENChar)*

PCENChar ::=
"-" | "." | [0-9] | "_" | [a-z] | #xB7 | [#xC0-#xD6] | [#xD8-#xF6] | [#xF8-#x37D] | [#x37F-#x1FFF] | [#x200C-#x200D] | [#x203F-#x2040] | [#x2070-#x218F] | [#x2C00-#x2FEF] | [#x3001-#xD7FF] | [#xF900-#xFDCF] | [#xFDF0-#xFFFD] | [#x10000-#xEFFFF]
```

Do we want to get this far?

Reply:
I don't have enough context to answer that. 😄 My guess would be "probably not".